### PR TITLE
Add imageService 'custom' to cloudflare integration

### PR DESCRIPTION
**Pull request by FlorianFlatscher**
## Changes

- Added a new imageService option to cloudflare integraion: `custom`
- Closes: #228 

## Testing

No Tests

## Docs

- https://github.com/withastro/docs/pull/7813

### CODE REVIEW

**Review by alexanderniebuhr — CHANGES_REQUESTED** [review]
@FlorianFlatscher Thanks for the contribution, this is actually something I wanted to add for months, but didn't have the time to do so. Very happy that you do it.

**This needs a Docs PR, once you created the PR in the docs repo, please cross-link them.** Your suggested docs are a good start, I would add some note that users need to make sure the custom image service is compatible with Cloudflare in production and that there are no guardlines or protections when using `imageService: 'custom'`

As for the code, that's a good start. Could you please also make sure the adapter options are updated? (https://github.com/withastro/adapters/blob/c22bb21f6bb0fece2bf29b9acaff08a78a6fce43/packages/cloudflare/src/index.ts#L20-L62)

It would also be great if you can add a changeset `pnpm exec changeset`, _(this would be a minor release)_.

Once those two things are addressed, I'm happy to do a final review and ship it.

**Comment by FlorianFlatscher** [comment]
Thank you for your awesome guidance on my first contribution!

- I created and linked a docs pr
- Also updated adapter options
- changeset is in-coming

File: .changeset/stupid-houses-laugh.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/cloudflare': minor
+---
+
+Add imageService 'custom' to cloudflare integration
```
Comment (alexanderniebuhr) [inline .changeset/stupid-houses-laugh.md:5]:
```suggestion
Adds a new option for the Image service to the Cloudflare adapter. `imageService: 'custom'` does use the user defined settings, without applying any modification to it. **You need to make sure that the configured settings are compatible with Cloudflare's `workerd` runtime yourself.**
```

**Review by alexanderniebuhr — APPROVED** [review]
Thanks for the contribution! ❤️

**Comment by alexanderniebuhr** [comment]
Actually I need to try something to see if it is actually working, before I can merge this.

**Pull request merged**